### PR TITLE
Add validation of scheduling policy

### PR DESCRIPTION
**Pull request by josephdrichard**

### CODE REVIEW

**Comment by josephdrichard** [comment]
@SchSeba @aneeshkp

File: api/v1/ptpconfig_types.go [hunk 1]
```diff
@@ -61,12 +61,15 @@ type PtpConfigList struct {
...
-	Ptp4lOpts             *string            `json:"ptp4lOpts,omitempty"`
-	Phc2sysOpts           *string            `json:"phc2sysOpts,omitempty"`
-	Ptp4lConf             *string            `json:"ptp4lConf,omitempty"`
-	PtpSchedulingPolicy   *string            `json:"ptpSchedulingPolicy,omitempty"`
+	Name        *string `json:"name"`
+	Interface   *string `json:"interface,omitempty"`
+	Ptp4lOpts   *string `json:"ptp4lOpts,omitempty"`
+	Phc2sysOpts *string `json:"phc2sysOpts,omitempty"`
+	Ptp4lConf   *string `json:"ptp4lConf,omitempty"`
+	// +kubebuilder:validation:Enum=SCHED_OTHER;SCHED_FIFO;
```
Comment (SchSeba) [inline api/v1/ptpconfig_types.go:69]: I think `SCHED_OTHER` is not valid here as there is nothing in the linuxptp

File: api/v1/ptpconfig_webhook.go [hunk 2]
```diff
@@ -118,6 +122,12 @@ func (r *PtpConfig) validate() error {
 		if summary_interval != logSyncInterval {
 			return errors.New("summary_interval " + summary_interval + " must match logSyncInterval " + logSyncInterval)
 		}
+
+		if profile.PtpSchedulingPolicy != nil && *profile.PtpSchedulingPolicy == "SCHED_FIFO" {
```
Comment (SchSeba) [inline api/v1/ptpconfig_webhook.go:126]: do you need to check also about `SCHED_OTHER`?

**Review by SchSeba — COMMENTED** [review]
can you please rebase on master we already merge one of the commits from this PR on a different one

File: api/v1/ptpconfig_types.go (hunk 1 above)
Comment (josephdrichard) [reply api/v1/ptpconfig_types.go:69]: SCHED_OTHER will have the same behaviour as just not setting it.

File: api/v1/ptpconfig_webhook.go (hunk 2 above)
Comment (josephdrichard) [reply api/v1/ptpconfig_webhook.go:126]: My opinion is that we don't.  If SCHED_OTHER, no validation is necessary.

**Pull request merged**